087-external-certificate-manager.md

```diff
@@ -0,0 +1,241 @@
+# External Certificate Manager
+
+This proposal aims to allow Strimzi users to use an external certificate manager, specifically [cert-manager](https://cert-manager.io/), to manage certificates.
+
+## Current situation
+
+There are two different categories of certificates that Strimzi handles:
+* The term "cluster" refers to certificates that are issued for the Strimzi components:
+  * ZooKeeper nodes
+  * Kafka nodes
+  * Cluster, User and Topic operators
+  * Cruise Control
+  * Kafka Exporter
+* The term "clients" refers to certificates that are issued for user applications using the User Operator, or through another external mechanism chosen by the user.
+
+For both categories, to provide a secure, TLS-enabled setup by default when deploying Kafka clusters, Strimzi integrated its own CA operations into the Cluster Operator.
+The Cluster Operator accomplishes this by using openssl to generate self-signed root CA certificates and private keys which it then uses to directly sign end-entity (EE) certificates.
+This CA certificate has zero pathlen, which means it cannot sign any intermediate CA.
+A cluster CA and clients CA are generated.
+These CAs are only used for Kafka clusters and a unique instance of each CA is used for each Kafka cluster.
+
+In addition to Strimzi fully managing the certificates as described above, there are options for users to partially manage the certificates:
+* Users can [install and use their own CA certificate and private keys](https://strimzi.io/docs/operators/latest/deploying#installing-your-own-ca-certificates-str), instead of using the defaults generated by the Cluster Operator.
+  When using this option, both the CA certificate and private key must be provided, and Strimzi still issues the end-entity (EE) certificates that are presented by the components.
+* Users can [provide custom listener certificates](https://strimzi.io/docs/operators/latest/deploying#proc-installing-certs-per-listener-str) for TLS encryption.
+  This option only affects how user applications connect to Kafka.
+  It does not change how the Strimzi components connect to Kafka, or how the Kafka brokers connect to each other.
+
+None of the existing options allow all end-entity certificates to be issued by a tool that isn't Strimzi.
```

Comment:
As mentioned in the comment above, this is not entirely true for the Clients CA where there are plenty users issuing their own user certificates.

For Cluster CA, users can in theory pre-create the certificate Secrets. But it is not documented and getting the SANs of the EE certificates right is challenging. I know some users _tried this_, but I'm not aware of anyone actually doing this in real environments.

Reply:
> For Cluster CA, users can in theory pre-create the certificate Secrets. But it is not documented and getting the SANs of the EE certificates right is challenging. I know some users tried this, but I'm not aware of anyone actually doing this in real environments.

For all these reasons I would not mentioned it here. I guess that referring to end-entity certificates, Kate is talking about the internal cluster components and not clients so the clients CA is out of the picture. Maybe we can clarify it here.

Reply:
I do not think it is clear here it is only about the Cluster CA.

Reply:
Here I meant all certificates, so those associated with both the Cluster CA and Clients CA. I realise that theoretically it is possible by pre-creating the certificate Secrets, but as you say this is challenging. Is this sentence needed? Or is the motivation below sufficient so that I could remove the sentence here?